salescustomeranalysis/src/main/java/MetricsCollector.java

```diff
@@ -0,0 +1,101 @@
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Collects analytical metrics over a list of orders.
+ * The instance holds a reference to the provided orders list.
+ */
+public class MetricsCollector {
+
+    private final List<Order> orders;
+
+    /**
+     * Creates a new metrics collector over the given orders.
+     *
+     * @param orders source orders to analyze; must be non-null
+     */
+    public MetricsCollector(List<Order> orders) {
+        this.orders = orders;
```

Comment:
there is no non-null validation, despite the Javadoc stating "must be non-null".